<launch>
  <arg name="debug" default="false"/>
  <arg name="dt" default="0.2"/>
  <arg name="max_num" default="-1"/>
  <!--<arg name="debug_valgrind" default="false"/>-->

  <arg name="prefix" value="" unless="$(arg debug)"/>
  <arg name="prefix" value="gdb -ex run --args" if="$(arg debug)"/>

  <node pkg="iarc7_planner"
    type="test_planner_voxel_map"
    name="test_planner"
    launch-prefix="$(arg prefix)"
    output="screen">
    <remap from="~pose" to="/move_base_simple/goal"/>
    <param name="topic" value="/cloud"/>

    <param name="goal_x" value="17"/>
    <param name="goal_y" value="15"/>
    <param name="goal_z" value="3"/>
    <!-- Set range -->
    <param name="origin_x" value="10.0"/>
    <param name="origin_y" value="10.0"/>
    <param name="origin_z" value="0.0"/>
    <param name="range_x" value="20"/>
    <param name="range_y" value="20"/>
    <param name="range_z" value="5"/>
    <!-- Set dynamic constraints -->
    <param name="dt" value="$(arg dt)"/>
    <param name="v_max" value="10.0"/>
    <param name="a_max" value="10"/>
    <param name="u_max" value="50.0"/>
    <!--<param name="t_max" value="$(eval dt*10)"/>-->
    <param name="t_max" value="-1"/>
    <param name="use_3d" value="false"/>
    <param name="use_prior" value="false"/>
    <param name="use_acc" value="false"/>
    <param name="use_jrk" value="false"/>
    <param name="num" value="3"/>
    <param name="w" value="10000"/>
    <param name="epsilon" value="2"/>
    <param name="max_num" value="$(arg max_num)"/>
    <param name="robot_r" value="0.5"/>
  </node>

</launch>
